services/QuillLMS/spec/models/teacher_activity_feed_spec.rb

```diff
@@ -0,0 +1,166 @@
+
+    describe '#data_for_activity_feed' do
+      it "has all activity sessions completed for that teacher's classroom, in reverse chronological order" do
+        activity_session
+
+        data = TeacherActivityFeed.get(teacher.id)
+        expect(data.length).to eq(3)
+        expect(data[0][:id]).to eq(activity_session.id)
+        expect(data[1][:id]).to eq(activity_session1.id)
+        expect(data[2][:id]).to eq(activity_session2.id)
+      end
+    end
+
+  end
+
+  describe '#text_for_score' do
+    let(:lesson_activity_session) { create(:lesson_activity_session) }
+    let(:diagnostic_activity_session) { create(:diagnostic_activity_session) }
+    let(:proficient_activity_session) { create(:connect_activity_session, percentage: 0.90) }
+    let(:nearly_proficient_activity_session) { create(:grammar_activity_session, percentage: 0.75) }
+    let(:not_yet_proficient_activity_session) { create(:proofreader_activity_session, percentage: 0.40) }
+
+    let(:feed) { TeacherActivityFeed.new('fake_id') }
+
+    context 'when the activity session is a lesson activity' do
+      it "returns #{ActivitySession::COMPLETED}" do
+        text = feed.send(:text_for_score, lesson_activity_session.classification.key, lesson_activity_session.percentage)
+
+        expect(text).to eq(ActivitySession::COMPLETED)
+      end
+    end
+
+    context 'when the activity session is a diagnostic activity' do
+      it "returns #{ActivitySession::COMPLETED}" do
+        text = feed.send(:text_for_score, diagnostic_activity_session.classification.key, diagnostic_activity_session.percentage)
+
+        expect(text).to eq(ActivitySession::COMPLETED)
+      end
+    end
+
+    context 'when the activity session percentage is at or above the proficiency cutoff' do
+      it "returns #{ActivitySession::PROFICIENT}" do
+        text = feed.send(:text_for_score, proficient_activity_session.classification.key, proficient_activity_session.percentage)
+
+        expect(text).to eq(ActivitySession::PROFICIENT)
+      end
+    end
+
+    context 'when the activity session percentage is at or above the nearly proficient cutoff' do
+      it "returns #{ActivitySession::NEARLY_PROFICIENT}" do
+        text = feed.send(:text_for_score, nearly_proficient_activity_session.classification.key, nearly_proficient_activity_session.percentage)
+
+        expect(text).to eq(ActivitySession::NEARLY_PROFICIENT)
+      end
+    end
+
+    context 'when the activity session percentage is below the nearly proficient cutoff' do
+      it "returns #{ActivitySession::NOT_YET_PROFICIENT}" do
+        text = feed.send(:text_for_score, not_yet_proficient_activity_session.classification.key, not_yet_proficient_activity_session.percentage)
+
+        expect(text).to eq(ActivitySession::NOT_YET_PROFICIENT)
+      end
+    end
+
+  end
+
+  describe '#text_for_completed' do
```

Comment:
I love that you got exhaustive with these cases.  It makes me happy!

Reply:
These are mostly @emilia-friedberg, she did a great job here!